CMakeLists.txt

```diff
@@ -0,0 +1,345 @@
+file(GLOB_RECURSE GAME_ASM_FILES
+	CONFIGURE_DEPENDS RELATIVE ${PROJECT_SOURCE_DIR}/source ${PROJECT_SOURCE_DIR}/source/*.s)
+
+# Build Functions
+function(build_resource)
+	# Set Variables to make this easier
+	# i.e SPRITE
+	set(RESOURCE_NAME ${ARGV0})
+	# i.e SPRITES
+	set(PLURAL ${RESOURCE_NAME}S)
+	# passed in dependencies, tool or the previous object
+	set(TARGET_DEPENDENCIES ${ARGV1})
+	# Set helped variables based on what resource we are exporting and building
+	if(${RESOURCE_NAME} MATCHES "SPRITE")
+		set(TOOL_NAME sprite-export)
+		set(FOLDER_NAME sprites/)
+		set(OUT1 .s)
+		set(OUT2 .hpp)
+	elseif(${RESOURCE_NAME} MATCHES "BACKGROUND")
+		set(TOOL_NAME background-export)
+		set(FOLDER_NAME backgrounds/)
+		set(OUT1 .s)
+		set(OUT2 .hpp)
+	elseif(${RESOURCE_NAME} MATCHES "FONT")
+		set(TOOL_NAME font-export)
+		set(FOLDER_NAME fonts/)
+		set(OUT1 .s)
+		set(OUT2 .hpp)
+	elseif(${RESOURCE_NAME} MATCHES "SETTING")
+		set(TOOL_NAME audio-export-settings)
+		set(FOLDER_NAME "")
+		set(OUT1 -settings.hpp)
+		set(OUT2 -settings.json)
+	elseif(${RESOURCE_NAME} MATCHES "SOUND")
+		set(TOOL_NAME sound-export)
+		set(FOLDER_NAME sounds/)
+		set(OUT1 .s)
+		set(OUT2 .hpp)
+	endif()
+	set(RESOURCE_FILES ${ARGV2})
+	set(IN2 ${ARGV3})
+
+	#make a directory in build folder for the output files
+	file(MAKE_DIRECTORY ${CMAKE_BINARY_DIR}/data/${FOLDER_NAME})
+
+	set(OBJECT_LIST_NAME ${PLURAL}_OBJECT_LIST)
+	set(OUT1_LIST_NAME ${PLURAL}_OUT1_LIST)
+	set(OUT2_LIST_NAME ${PLURAL}_OUT2_LIST)
+
+	foreach(FILE ${RESOURCE_FILES})
+		get_filename_component(file_dir "${FILE}" PATH)
+		string(REPLACE "." ";." file ${FILE})
+		list(GET file 0 file_name)
+		list(GET file 1 file_type)
+
+		set(OUTPUT_FILE_1 ${CMAKE_BINARY_DIR}/data/${FOLDER_NAME}${file_name}${OUT1})
+		set(OUTPUT_FILE_2 ${CMAKE_BINARY_DIR}/data/${FOLDER_NAME}${file_name}${OUT2})
+
+		# This command generates the files from the source/data files
+		# OUTPUT means the files are set as Generated in cmake,
+		# so when they are used as a source, a dependency is set on their generation
+		# We pass in the dependencies to DEPENDS, the first call of this function passes in the Tools project as a dependency,
+		# so that all subsequent commands and targets are implicitely dependent on Tools having finished
+		# WORKING_DIRECTORY is set to the project root (${CMAKE_SOURCE_DIR}) so a relative path to the file that is to be exported
+		# is sent to the tool, for proper generation of namespaces
+		add_custom_command(OUTPUT ${OUTPUT_FILE_1} ${OUTPUT_FILE_2}
+			DEPENDS ${TARGET_DEPENDENCIES}
+			COMMAND ${CMAKE_BINARY_DIR}/tools/tools ${TOOL_NAME}
+				./data/${FOLDER_NAME}${file_name}${file_type}
+				${OUTPUT_FILE_1} ${OUTPUT_FILE_2} ${IN2}
+			WORKING_DIRECTORY ${CMAKE_SOURCE_DIR}
+			COMMENT "Exporting ${RESOURCE_NAME} ${file_name}")
+
+		# We don't want to generate and object and dependency? file for audio.json, since it only outputs a header and another json file
```

Comment:
The `.d` file is a "makefile" that simply declares all dependencies that the generated file has, as explained [here](https://stackoverflow.com/questions/11855386/using-g-with-mmd-in-makefile-to-automatically-generate-dependencies). Since the generated assembly files don't have any dependencies, the .d file might actually be omitted, because it's only relevant for the old Makefile build.

Also, what is stopping you from simply adding the `.s` files to the ROM final build list?

Reply:
Nothing stopping adding the .s files I suppose. I'll get rid of the .d stuff too since it isn't needed.